### PR TITLE
Add support for Password Generator ID's

### PR DESCRIPTION
**Pull request by colombeen**
```powershell
PS C:\> New-RandomPassword -PolicyID 2

Password
--------------
sdfQ@sfd54Adff!(§221aB

PS C:\> 
```

### CODE REVIEW

File: functions/New-RandomPassword.ps1 [hunk 1]
```diff
@@ -1,103 +1,114 @@
...
+              $includeuppercase = $true
             }
+            [string]$includebrackets = ([string]$includebrackets).ToLower()
+            [string]$includelowercase = ([string]$includelowercase).ToLower()
+            [string]$includeuppercase = ([string]$includeuppercase).ToLower()
+            [string]$includespecialcharacters = ([string]$includespecialcharacters).ToLower()
+            [string]$includenumbers = ([string]$includenumbers).ToLower()
+
+            $uri = "/api/generatepassword/?IncludeAlphaSpecial=true&IncludeWordPhrases=false&minLength=$length&maxLength=$length&lowerCaseChars=$includelowercase&upperCaseChars=$includeuppercase&numericChars=$includenumbers&higherAlphaRatio=true&ambiguousChars=true&specialChars=$includespecialcharacters&specialCharsText=!$%+-_=^&bracketChars=$includebrackets&bracketCharsText={}()&NumberOfWords=0&MaxWordLength=0&PrefixAppend=P&SeparateWords=N&ExcludeChars=$excludedcharacters&GeneratePattern=false&Pattern=null&Qty=$quantity"
+            Write-Verbose "[$(Get-Date -format G)] [GET] $uri"
```
Comment (dnewsholme) [inline functions/New-RandomPassword.ps1]:
```suggestion
            
```

Comment (dnewsholme) [inline functions/New-RandomPassword.ps1]:
```suggestion

```

File: functions/New-RandomPassword.ps1 [hunk 2]
```diff
@@ -1,103 +1,114 @@
...
+            Get-PasswordStateResource -uri $uri
+          }
         }
-
+      }
+      # Generate a password using an existing Password Generator ID
+      'PolicyID' {
+        $uri = "/api/generatepassword/?PasswordGeneratorID=$PolicyID"
+        Write-Verbose "[$(Get-Date -format G)] [GET] $uri"
```
Comment (dnewsholme) [inline functions/New-RandomPassword.ps1]:
```suggestion
```

File: functions/New-RandomPassword.ps1 [hunk 3]
```diff
@@ -1,103 +1,114 @@
...
+        Get-PasswordStateResource -uri $uri
```
Comment (dnewsholme) [inline functions/New-RandomPassword.ps1]:
```suggestion
```

File: functions/New-RandomPassword.ps1 [hunk 4]
```diff
@@ -1,103 +1,114 @@
...
+      }
     }
+  }
```
Comment (dnewsholme) [inline functions/New-RandomPassword.ps1]:
```suggestion
    Write-Verbose "[$(Get-Date -format G)] [GET] $uri"
    Get-PasswordStateResource -uri $uri
  }
```

**Comment by colombeen** [comment]
I'm not entirely sure about the name for the parameter

**Comment by dnewsholme** [comment]
I've added a couple of suggestions to clean up section of the code so there isn't duplicate lines.

Could probably do with adding a pester test for the new functionality as well.

Adding the following to New-RandomPassword.Tests.ps1 at line 14 should cover it.

`    It "Generates a Password from a policy" {
        (New-RandomPassword -PolicyID 1).Password | Should -not -BeNullOrEmpty
    }`

**Pull request merged**